dpnp/backend/extensions/lapack/gesv.cpp

```diff
@@ -273,6 +273,162 @@ std::pair<sycl::event, sycl::event>
 }
 
+std::pair<sycl::event, sycl::event>
+    gesv_batch(sycl::queue &exec_q,
+               dpctl::tensor::usm_ndarray coeff_matrix,
+               dpctl::tensor::usm_ndarray dependent_vals,
+               const std::vector<sycl::event> &depends)
+{
+    const int coeff_matrix_nd = coeff_matrix.get_ndim();
+    const int dependent_vals_nd = dependent_vals.get_ndim();
+
+    if (coeff_matrix_nd != 3) {
+        throw py::value_error("The coefficient matrix has ndim=" +
+                              std::to_string(coeff_matrix_nd) +
+                              ", but a 3-dimensional array is expected.");
+    }
+
+    if (dependent_vals_nd < 2 || dependent_vals_nd > 3) {
+        throw py::value_error(
+            "The dependent values array has ndim=" +
+            std::to_string(dependent_vals_nd) +
+            ", but a 2-dimensional or 3-dimensional array is expected.");
+    }
+
+    const py::ssize_t *coeff_matrix_shape = coeff_matrix.get_shape_raw();
+    const py::ssize_t *dependent_vals_shape = dependent_vals.get_shape_raw();
+
+    // The coeff_matrix and dependent_vals arrays must be F-contiguous arrays
+    // with the shapes (n,n,batch_size) and (n,nrhs,batch_size) or
+    // (n,batch_size) respectively
+    if (coeff_matrix_shape[0] != coeff_matrix_shape[1]) {
+        throw py::value_error("The coefficient matrix must be square,"
+                              " but got a shape of (" +
+                              std::to_string(coeff_matrix_shape[0]) + ", " +
+                              std::to_string(coeff_matrix_shape[1]) + ").");
+    }
+
+    if (coeff_matrix_shape[0] != dependent_vals_shape[0]) {
+        throw py::value_error("The first dimension (n) of coeff_matrix and"
+                              " dependent_vals must be the same, but got " +
+                              std::to_string(coeff_matrix_shape[0]) + " and " +
+                              std::to_string(dependent_vals_shape[0]) + ".");
+    }
+
+    if (dependent_vals_nd == 2) {
+        if (coeff_matrix_shape[2] != dependent_vals_shape[1]) {
+            throw py::value_error(
+                "The batch_size of "
+                " coeff_matrix and dependent_vals must be"
+                " the same, but got " +
+                std::to_string(coeff_matrix_shape[2]) + " and " +
+                std::to_string(dependent_vals_shape[1]) + ".");
+        }
+    }
+    else if (dependent_vals_nd == 3) {
+        if (coeff_matrix_shape[2] != dependent_vals_shape[2]) {
+            throw py::value_error(
+                "The batch_size of "
+                " coeff_matrix and dependent_vals must be"
+                " the same, but got " +
+                std::to_string(coeff_matrix_shape[2]) + " and " +
+                std::to_string(dependent_vals_shape[2]) + ".");
+        }
+    }
+
+    // check compatibility of execution queue and allocation queue
+    if (!dpctl::utils::queues_are_compatible(exec_q,
```

Comment:
Can we have common checks decoupled to single internal function to reduce code duplication?

Reply:
Done!